### PR TITLE
Lf 528 reset password success modal

### PR DESCRIPTION
**Pull request by JimmyRowland**

### CODE REVIEW

**Review by kaavyalakshmanan — APPROVED** [review]

File: packages/api/src/templates/password_reset_email.html [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  LiteFarm is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ -->
+
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>LiteFarm Farm Access Restored Email</title>
```
Comment (beetai) [inline packages/api/src/templates/password_reset_email.html:19]: Incorrect email subject line

File: packages/api/src/templates/password_reset_email.html [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        .button {
+            -webkit-appearance: button!important;
+            -moz-appearance: button!important;
+
+            text-decoration: none!important;
+            color: #282B36 !important;
+            background: #FCE38D!important;
+            box-shadow: 0px 2px 8px rgba(102, 115, 138, 0.3)!important;
+            border-radius: 4px!important;
+            padding: 12px 100px!important;
```
Comment (beetai) [inline packages/api/src/templates/password_reset_email.html:77]: I've made a change to the padding and width of the button, but this has since been reverted. Please restore the change.

Comment (beetai) [reply packages/api/src/templates/password_reset_email.html:77]: Image from iOS

File: packages/api/src/templates/password_reset_email.html [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+            text-align: center!important;
+            margin-top: 34px!important;
+        }
+
+        .footer p {
+            font-style: normal!important;
+            font-weight: normal!important;
+            font-size: 12px!important;
+            line-height: 24px!important;
+            color: #9FAABE!important;
```
Comment (beetai) [inline packages/api/src/templates/password_reset_email.html:99]: Apply consistent indentation (this also applies to all other templates)

File: packages/api/src/controllers/passwordResetController.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        // generate token for logging user in
+        const id_token = await createAccessToken({ user_id });
+
+        const template_path = '../templates/reset_password_confirmation.html';
+        const subject = 'Your LiteFarm password has been changed';
+        const replacements = {
+          first_name,
+        };
+        const sender = 'system@litefarm.org';
+        await emailSender.sendEmail(template_path, subject, replacements, email, sender, `/?email=${email}`);
```
Comment (beetai) [inline packages/api/src/controllers/passwordResetController.js:126]: There is a flaw with the URL that is being sent in the email. If users use a `+` symbol in email for their LiteFarm account, the email will not be displayed correctly when the field in automatically populated.

File: packages/api/package.json [hunk 5]
```diff
@@ -44,6 +44,7 @@
     "chai": "^4.2.0",
     "chai-http": "^4.3.0",
     "cors": "^2.8.5",
+    "crypto-js": "^4.0.0",
```
Comment (beetai) [inline packages/api/package.json:47]: Did we end up using `crypto-js`? I imagine we should clear this out if we did not

File: packages/api/src/controllers/passwordResetController.js [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+          reset_token_version = 1;
+          created_at = sendEmailDate;
+        } else if (reset_token_version === 3) {
+          return res.status(400).send('Reached maximum number of available reset tokens');
+        } else {
+          reset_token_version++;
+        }
+
+        // generate token
+        // payload: user_id, reset_token_version, email, first_name
```
Comment (beetai) [inline packages/api/src/controllers/passwordResetController.js:55]: alter comment or remove

File: packages/api/src/controllers/passwordResetController.js [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+      } catch (error) {
+        console.log(error);
+        return res.status(400).json(error);
+      }
+    };
+  }
+
+  static validateToken() {
+    return async (req, res) => {
+      // passwordResetController.isTokenValid()
```
Comment (beetai) [inline packages/api/src/controllers/passwordResetController.js:91]: remove comment

File: packages/api/src/controllers/passwordResetController.js [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+      return res.status(200).json({ isValid: true });
+    };
+  }
+
+  static resetPassword() {
+    return async (req, res) => {
+      // if(passwordResetController.isTokenValid())
```
Comment (beetai) [inline packages/api/src/controllers/passwordResetController.js:98]: remove comment

File: packages/api/src/util/generateKeypair.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
+/**
+ * This module will generate a public and private keypair and save to current directory
+ *
+ * Make sure to save the private key elsewhere after generated!
+ */
+const crypto = require('crypto');
+
+function genKeyPair() {
```
Comment (beetai) [inline packages/api/src/util/generateKeypair.js:8]: Remove this file

File: packages/webapp/.gitignore [hunk 10]
```diff
@@ -25,4 +25,6 @@ yarn-error.log*
 ./cypress/videos*
 ./cypress/integration/__snapshots__
 
-src/lang/*/*_old.json
\ No newline at end of file
+src/lang/*/*_old.json
+
+packages/.DS_Store
```
Comment (beetai) [inline packages/webapp/.gitignore:30]: Should we move this ignore to root level, to ignore all `.DS_Store`?

File: packages/webapp/src/Routes.js [hunk 11]
```diff
@@ -383,7 +384,9 @@ const Routes = () => {
   } else if (!isAuthenticated()) {
     return (
       <Switch>
-        {/*<Route path="/callback" render={(props) => {*/}
+        {/* TODO: CREATE RESET PASSWORD COMPONENT (on /callback) TO DISPATCH ACTION FOR VALIDATING TOKEN */}
```
Comment (beetai) [inline packages/webapp/src/Routes.js:387]: Remove TODO comment

File: packages/api/src/middleware/acl/checkResetPasswordTokenContent.js [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+const passwordModel = require('./../../models/passwordModel');
+async function checkResetPasswordTokenContent(req, res, next) {
+  const { user_id, created_at } = req.user;
+  const result = await passwordModel.query().select('created_at').where('user_id', user_id).first();
+  if(result.created_at.getTime() <= created_at) {
+    next();
+  }  else {
```
Comment (beetai) [inline packages/api/src/middleware/acl/checkResetPasswordTokenContent.js:22]: A little nitpicky, but standardize the spacings around `if` and `else`

**Review by beetai — CHANGES_REQUESTED** [review]
final push

File: packages/api/src/templates/password_reset_email.html (hunk 2 above)
Comment (beetai) [reply packages/api/src/templates/password_reset_email.html]: New changes have not fixed the issue I showed above. Let me know how the email looks on mobile devices you have access to.
I suggest you use what was put previously:
padding: 12px 0 !important;
width: 220px !important;

File: packages/api/src/templates/sendEmailTemplate.js [hunk 13]
```diff
@@ -21,8 +21,19 @@ const path = require('path');
 const jsdom = require('jsdom');
 const { JSDOM } = jsdom;
 
+const emails = {
+  INVITATION: { subject: '', path: 'invitation_to_farm_email.html' },
+  CONFIRMATION: { subject: '', path: 'send_confirmation_email.html' },
+  WITHHELD_CONSENT: { subject: '', path: 'withheld_consent_email.html' },
+  ACCESS_RESTORE: { subject: '', path: 'restoration_of_access_to_farm_email.html' },
+  ACCESS_REVOKE: { subject: '', path: 'revocation_of_access_to_farm_email.html' },
```
Comment (beetai) [inline packages/api/src/templates/sendEmailTemplate.js:29]: Add subject lines

**Review by beetai — CHANGES_REQUESTED** [review]

File: packages/api/src/templates/invitation_to_farm_email.html [hunk 14]
```diff
@@ -15,133 +15,132 @@
...
       }
 
       .button-aligner {
-        text-align: center!important;
+          text-align: center !important;
       }
 
       .button {
-        -webkit-appearance: button!important;
-        -moz-appearance: button!important;
```
Comment (beetai) [inline packages/api/src/templates/invitation_to_farm_email.html:70]: Why did you remove these? We might not need to remove any border if we use this

File: packages/api/src/templates/sendEmailTemplate.js (hunk 13 above)
Comment (kikemarquez8) [reply packages/api/src/templates/sendEmailTemplate.js:29]: This belongs to my next PR, so nvm this for now.
